test/unit/TicketBroker.js

```diff
@@ -620,4 +648,245 @@ contract("TicketBroker", accounts => {
+            })
+        })
+
+        it("emits an Unlock event indexed by sender", async () => {
+            const fromBlock = (await web3.eth.getBlock("latest")).number
+            await broker.fundDeposit({from: sender, value: 1000})
+
+            await broker.unlock({from: sender})
+
+            const events = await broker.getPastEvents("Unlock", {
+                filter: {
+                    sender
+                },
+                fromBlock,
+                toBlock: "latest"
+            })
+            assert.equal(events.length, 1)
+            const event = events[0]
+            assert.equal(event.returnValues.sender, sender)
+        })
+    })
+
+    describe("cancelUnlock", () => {
+        it("reverts if sender is not in an unlocking state", async () => {
+            await expectRevertWithReason(broker.cancelUnlock(), "no unlock request in progress")
+        })
+
+        it("sets isUnlockInProgress to false", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+            await broker.unlock()
+
+            await broker.cancelUnlock()
+
+            const isUnlockInProgress = await broker.isUnlockInProgress.call(sender)
+            assert.equal(isUnlockInProgress, false)
+        })
+
+        it("prevents withdrawal", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+            await broker.unlock()
+            await fixture.rpc.wait(unlockPeriod)
+
+            await broker.cancelUnlock()
+
+            await expectRevertWithReason(broker.withdraw(), "account is locked")
+        })
+
+        it("emits an UnlockCancelled event", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+            await broker.unlock()
+
+            const txResult = await broker.cancelUnlock()
+
+            truffleAssert.eventEmitted(txResult, "UnlockCancelled", ev => {
+                return ev.sender === sender
+            })
+        })
+
+        it("emits an UnlockCancelled event with an indexed sender", async () => {
+            await broker.fundDeposit({from: sender, value: 1000})
+            const fromBlock = (await web3.eth.getBlock("latest")).number
+            await broker.unlock()
+
+            await broker.cancelUnlock()
+
+            const events = await broker.getPastEvents("UnlockCancelled", {
+                filter: {
+                    sender
+                },
+                fromBlock,
+                toBlock: "latest"
+            })
+            assert.equal(events.length, 1)
+            const event = events[0]
+            assert.equal(event.returnValues.sender, sender)
+        })
+    })
+
+    describe("withdraw", () => {
+        it("reverts when both deposit and penaltyEscrow are zero", async () => {
+            await expectRevertWithReason(broker.withdraw(), "sender deposit and penalty escrow are zero")
+        })
+
+        it("reverts when account is locked", async () => {
```

Comment:
Perhaps we could split this test into 2 tests: the first for when an unlock request was never made and the second for when an unlock request was made, but the sender has not waited through the unlock period yet

Reply:
👍